### PR TITLE
[API-1062] Add Duplicate Pentest endpoint. Update changelog.

### PR DESCRIPTION
**Pull request by zmomen-cobalt**
Ticket: [API-1062](https://zombie.atlassian.net/browse/API-1062)

Please review the paragraph text for "duplicate a pentest."  

**Highlights**
- the pentest id has to be a valid ID. 
- the pentest being duplicated has to belong to the org in the x-org-token
- the newly duplicated pentest will be created in Draft state with the same "settings" as the one being duplicated. 
- there is **no json** body for this POST endpoint. 
- I updated the changelog also.

### CODE REVIEW

File: versions/v2/content/pentests.md [hunk 1]
```diff
@@ -325,3 +325,30 @@ issue.
...
+curl -X POST "https://api.cobalt.io/pentests/YOUR-PENTEST-IDENTIFIER/duplicate" \
+  -H "Accept: application/vnd.cobalt.v2+json" \
+  -H "Authorization: Bearer YOUR-PERSONAL-API-TOKEN" \
+  -H "X-Org-Token: YOUR-V2-ORGANIZATION-TOKEN"
+```
+
+This endpoint duplicates a pentest based on the identifier provided and creates a new pentest.
+The pentest identifier has to be a valid pentest that belong to the organization specified in the `X-Org-Token` header.
+Note there is **no JSON body** required for this endpoint.
+The new pentest will be in **Draft state** and will have the same settings as the pentest provided in the identifier.
```
Comment (zmomen-cobalt) [inline versions/v2/content/pentests.md:341]: not a fan of "settings". Any other ideas?

Comment (grayson-cobalt) [reply versions/v2/content/pentests.md:341]: I think the right business domain word is "brief" ?

File: .gitignore [hunk 2]
```diff
@@ -12,3 +12,5 @@ node_modules/
 
 # netlify CLI tmp folder
 .netlify/
+
+.idea/
```
Comment (inikolov89) [inline .gitignore:16]: offtopic: markdown is the only place I think idea sucks :D

File: versions/v2/content/pentests.md [hunk 3]
```diff
@@ -325,3 +325,30 @@ issue.
...
+
+```sh
+curl -X POST "https://api.cobalt.io/pentests/YOUR-PENTEST-IDENTIFIER/duplicate" \
+  -H "Accept: application/vnd.cobalt.v2+json" \
+  -H "Authorization: Bearer YOUR-PERSONAL-API-TOKEN" \
+  -H "X-Org-Token: YOUR-V2-ORGANIZATION-TOKEN"
+```
+
+This endpoint duplicates a pentest based on the identifier provided and creates a new pentest.
+The pentest identifier has to be a valid pentest that belong to the organization specified in the `X-Org-Token` header.
```
Comment (inikolov89) [inline versions/v2/content/pentests.md:339]: This line is a bit weird to me, not sure if not being a native speaker, but `pentest identifier has to be a valid pentest` does not sound right to me.
```suggestion
The pentest identifier has to point to a valid pentest that belongs to the organization specified in the `X-Org-Token` header.
```

File: versions/v2/content/pentests.md [hunk 4]
```diff
@@ -325,3 +325,30 @@ issue.
...
+The new pentest will be in **Draft state** and will have the same settings as the pentest provided in the identifier.
+
+### HTTP Request
+
+`POST https://api.cobalt.io/pentests/YOUR-PENTEST-IDENTIFIER-HERE/duplicate`
+
+### Response
+
+On success, a `201` response code will be returned. A response header, `Location`, will contain the URL
+within Cobalt's API of the new duplicated pentest.
```
Comment (inikolov89) [inline versions/v2/content/pentests.md:350]: `new duplicated pentest` can we just say `new pentest` for simplicity? We are already in the duplicate context.

File: versions/v2/content/pentests.md [hunk 5]
```diff
@@ -325,3 +325,30 @@ issue.
...
+
+<aside class="notice">
+Remember - you can only request a pentest scoped to the organization specified in the <code>X-Org-Token</code> header.
```
Comment (inikolov89) [inline versions/v2/content/pentests.md:353]:
```suggestion
Remember - you can only duplicate a pentest within the organization specified in the <code>X-Org-Token</code> header.
```
just a suggestion to simplify.

File: versions/v2/content/pentests.md [hunk 6]
```diff
@@ -325,3 +325,30 @@ issue.
...
+## Duplicate a Pentest
+
+```sh
+curl -X POST "https://api.cobalt.io/pentests/YOUR-PENTEST-IDENTIFIER/duplicate" \
+  -H "Accept: application/vnd.cobalt.v2+json" \
+  -H "Authorization: Bearer YOUR-PERSONAL-API-TOKEN" \
+  -H "X-Org-Token: YOUR-V2-ORGANIZATION-TOKEN"
+```
+
+This endpoint duplicates a pentest based on the identifier provided and creates a new pentest.
```
Comment (nathansandi) [inline versions/v2/content/pentests.md:338]: I would suggest adding in the document, which Pentests states you can duplicate.

File: versions/v2/content/pentests.md [hunk 7]
```diff
@@ -325,3 +325,30 @@ issue.
...
+
+## Duplicate a Pentest
+
+```sh
+curl -X POST "https://api.cobalt.io/pentests/YOUR-PENTEST-IDENTIFIER/duplicate" \
+  -H "Accept: application/vnd.cobalt.v2+json" \
+  -H "Authorization: Bearer YOUR-PERSONAL-API-TOKEN" \
+  -H "X-Org-Token: YOUR-V2-ORGANIZATION-TOKEN"
+```
+
```
Comment (nathansandi) [inline versions/v2/content/pentests.md:337]: [we align it on slack] Just to formalize, to map in which states is possible to duplicate pentest :)

Comment (zmomen-cobalt) [reply versions/v2/content/pentests.md:337]: I did a full test locally and the only state that you **cannot duplicate** is "new". All other states are valid for duplication. I will update the documentation.

File: versions/v2/content/pentests.md (hunk 5 above)
Comment (zmomen-cobalt) [reply versions/v2/content/pentests.md]: updated here [c2f8d5c](https://github.com/cobalthq/cobalt-api-docs/pull/105/commits/c2f8d5cef49779c2a80f162d1116f47f9d82a66b)

File: versions/v2/content/pentests.md (hunk 4 above)
Comment (zmomen-cobalt) [reply versions/v2/content/pentests.md]: updated here [c2f8d5c](https://github.com/cobalthq/cobalt-api-docs/pull/105/commits/c2f8d5cef49779c2a80f162d1116f47f9d82a66b)

File: versions/v2/content/pentests.md (hunk 3 above)
Comment (zmomen-cobalt) [reply versions/v2/content/pentests.md]: updated here [c2f8d5c](https://github.com/cobalthq/cobalt-api-docs/pull/105/commits/c2f8d5cef49779c2a80f162d1116f47f9d82a66b)

File: versions/v2/content/pentests.md [hunk 8]
```diff
@@ -325,3 +325,31 @@ issue.
...
+This endpoint duplicates a pentest based on the identifier provided and creates a new pentest.
```
Comment (grayson-cobalt) [inline versions/v2/content/pentests.md:338]: I'm thinking the wording "This endpoint duplicates a pentest based on the identifier provided **and creates a new pentest**." is strange. To me, duplication implies creation of a new pentest. The "and creates a new pentest" implies two actions.

File: versions/v2/content/pentests.md [hunk 9]
```diff
@@ -325,3 +325,31 @@ issue.
...
+The pentest identifier has to point to a valid pentest that belongs to the organization specified in the `X-Org-Token` header.
```
Comment (grayson-cobalt) [inline versions/v2/content/pentests.md:339]: I think this wording could be better too "identifier has to point to". 
Perhaps something like:
"The pentest to be duplicated must exist and belong to the organization specified in the `X-Org-Token` header.

File: versions/v2/content/pentests.md [hunk 10]
```diff
@@ -325,3 +325,31 @@ issue.
...
+Note there is **no JSON body** required for this endpoint.
```
Comment (grayson-cobalt) [inline versions/v2/content/pentests.md:340]: "Request body" instead of "JSON body"? 🤔

File: versions/v2/content/pentests.md (hunk 8 above)
Comment (zmomen-cobalt) [reply versions/v2/content/pentests.md:338]: Yeah. I am changing it to

This endpoint creates a duplicate pentest based on the identifier provided.

**Review by grayson-cobalt — APPROVED** [review]

**Review by msdundar — APPROVED** [review]

**Review by nathansandi — APPROVED** [review]

**Comment by mjang-cobalt** [comment]
Thank you for including us on your API reference review. I'm sorry that we didn't have time to include our review before you merged this PR.  I see from the discussion in this PR, you care deeply about documentation, and that's great. We on the Cobalt Product Docs team want to help. 

Our time is limited, so we may not always be able to help in timeframe that you need. We recommend the following tool:

- [Vale](https://github.com/cobalthq/cobalt-product-public-docs/blob/main/GrammarLinter.md). In our product documentation, we've incorporated Vale to
  - Review our compliance with Cobalt styles, based in part on the [Google Developer Style Guide](https://developers.google.com/style).
    - If you use the Vale plugin with your IDE, you'll see links to explanations of "why," such as [Abbreviations not to
use](https://developers.google.com/style/abbreviations#dont-use). 
      - I use Vale with vsCode and at the Command Line. [Learn more.](https://github.com/cobalthq/cobalt-product-public-docs/blob/main/GrammarLinter.md#use-vale-in-your-ide)
  - Review readability. We want documentation that's more [readable](https://zombie.atlassian.net/wiki/spaces/PRODUCT/pages/2362671319/Readability+Standards) [than](url) the competition
    - I've noticed that most of our current API reference documentation files already meet Flesch-Kincaid standards for readability. Congratulations! (We reserve the right to make additional readability suggestions.)

Note: Vale has already been approved by the Cobalt security team.

**Pull request merged**